sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareDirectoryGetFilesAndDirectoriesOptions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.Files.Shares.Models
+{
+    /// <summary>
+    /// Optional parameters for <see cref="ShareDirectoryClient.GetFilesAndDirectoriesAsync(ShareDirectoryGetFilesAndDirectoriesOptions, System.Threading.CancellationToken)"/>.
+    /// </summary>
+    public class ShareDirectoryGetFilesAndDirectoriesOptions
+    {
+        /// <summary>
+        /// Optional string that filters the results to return only
+        /// files and directories whose name begins with the specified prefix.
+        /// </summary>
+        public string Prefix { get; set; }
+
+        /// <summary>
+        /// Optional.  Specifies traits to include in the <see cref="ShareFileItem"/>.
+        /// </summary>
+        public ShareFileTraits? Traits { get; set; }
+
+        /// <summary>
+        /// Optional.  Specified that extended info should be included in the <see cref="ShareFileItem"/>.
+        /// </summary>
+        public bool? IncludeExtendedInfo { get; set; }
```

Comment:
Why an optional bool? I'm not sure what this means as a user and I'm not sure what we gain by knowing whether they set the value or not.

Reply:
+1

Reply:
Because the default value for regular bool is false.  I don't want to send include extended info = false by default to the service, because I think it is an invalid input when "include" is specified.  [spec](https://microsoft.sharepoint.com/:w:/t/AzureStorage/EWClFgIBx3VMhPSrpcg98SIBPuLvbfVVOEZDUM-JShArDA?e=uZKj63)